src/wp-includes/embed.php

```diff
@@ -356,11 +356,23 @@ function wp_oembed_add_discovery_links() {
 /**
  * Adds the necessary JavaScript to communicate with the embedded iframes.
  *
+ * This function is no longer used directly. For back-compat it exists exclusively as a way to indicate that the oEmbed
+ * host JS _should_ be added. In `default-filters.php` there remains this code:
+ *
+ *     add_action( 'wp_head', 'wp_oembed_add_host_js' )
+ *
+ * Historically a site has been able to disable adding the oEmbed host script by doing:
+ *
+ *     remove_action( 'wp_head', 'wp_oembed_add_host_js' )
+ *
+ * In order to ensure that such code still works as expected, this function remains. There is now a `has_action()` check
+ * in `wp_maybe_enqueue_oembed_host_js()` to see if `wp_oembed_add_host_js()` has not been unhooked from running at the
+ * `wp_head` action.
+ *
  * @since 4.4.0
+ * @see wp_maybe_enqueue_oembed_host_js()
  */
-function wp_oembed_add_host_js() {
-	add_filter( 'embed_oembed_html', 'wp_maybe_enqueue_oembed_host_js' );
-}
+function wp_oembed_add_host_js() {}
```

Comment:
Worth deprecating?

Reply:
Humm. If so, them it could only be soft-`@deprecated`. We couldn't add `_deprecated_function()` as otherwise every site will have a deprecation warning happening at `wp_head`.

Reply:
I agree with adding a soft `@deprecated 5.9.0` for documentation purposes. But as @westonruter points out, can't do a hard deprecation. Added suggestion and will merge it into the PR.

Reply:
Soft deprecated in https://github.com/WordPress/wordpress-develop/pull/2059/commits/02f599966343c546a1d6b8b38fabccb77ebcdebc.